src/mc_control/samples/Impedance/mc_impedance_controller.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015-2020 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+#include "mc_impedance_controller.h"
+
+#include <mc_rtc/constants.h>
+
+namespace mc_control
+{
+
+MCImpedanceController::MCImpedanceController(std::shared_ptr<mc_rbdyn::RobotModule> robot_module, double dt)
+: MCController(robot_module, dt)
+{
+  qpsolver->addConstraintSet(contactConstraint);
+  qpsolver->addConstraintSet(kinematicsConstraint);
+  qpsolver->addConstraintSet(selfCollisionConstraint);
+  qpsolver->addConstraintSet(*compoundJointConstraint);
+
+  // add PostureTask
+  qpsolver->addTask(postureTask);
+  postureTask->stiffness(1);
+  postureTask->weight(1);
+
+  // add EndEffectorTask for root link
+  rootLinkTask =
+      std::make_shared<mc_tasks::EndEffectorTask>("base_link", robots(), robots().robotIndex(), 100.0, 10000.0);
+  rootLinkTask->set_ef_pose(sva::PTransformd(Eigen::Vector3d(0, 0, 1)));
+  solver().addTask(rootLinkTask);
```

Comment:
> Why a task on the root link?

Not so much reason. Just to make a example with most simple settings, the arm of JVRC1 is considered as a manipulator fixed to the ground.

Instead of it, adding foot/floor contact, dynamicsConstraint, CoM task is also good for humanoid example.